lib/kitchen/driver/ec2.rb

```diff
@@ -201,7 +201,6 @@ def create(state) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
         wait_until_ready(server, state)
 
         info("EC2 instance <#{state[:server_id]}> ready.")
-        state[:hostname] = hostname(server)
```

Comment:
The problem is that I reassign the `server` object passed into `wait_until_ready`, so the server object here refers to the first iteration of it (before we continually updated it by querying for its hostname)